migrations/2_deploy_contracts.js

```diff
@@ -5,25 +5,37 @@ var Storage = artifacts.require("./contracts/satelites/Storage.sol");
 var Controller = artifacts.require("./contracts/controller/Controller.sol");
 
 module.exports = function(deployer, network, accounts) {
-  let controller;
-  deployer.deploy([Nutz, Power, PullPayment, Storage]).then(function() {
-    return deployer.deploy(Controller, Power.address, PullPayment.address, Nutz.address, Storage.address);
+  let power, pull, nutz, storage;
+  deployer.deploy(Nutz)
+  .then(function() {
+    return deployer.deploy(Power);
+  }).then(function() {
+    return deployer.deploy(PullPayment);
+  }).then(function() {
+    return deployer.deploy(Storage);
   }).then(function() {
-    return Controller.deployed();
-  }).then(function(_controller){
-    controller = _controller;
     return Nutz.deployed();
-  }).then(function(nutz) {
-    nutz.transferOwnership(controller.address);
+  }).then(function(_nutz) {
+    nutz = _nutz;
     return Power.deployed();
-  }).then(function(power) {
-    power.transferOwnership(controller.address);
+  }).then(function(_power) {
+    power = _power;
     return PullPayment.deployed();
-  }).then(function(pull) {
-    pull.transferOwnership(controller.address);
+  }).then(function(_pull) {
+    pull = _pull;
     return Storage.deployed();
-  }).then(function(storage) {
+  }).then(function(_storage) {
+    storage = _storage;
+    return deployer.deploy(Controller, power.address, pull.address, nutz.address, storage.address);
+  }).then(function() {
+    return Controller.deployed();
+  }).then(function(controller){
+    console.log('Transfering satellites\' ownership to Controller..');
+    nutz.transferOwnership(controller.address);
```

Comment:
probably won't work without sendTransaction

Reply:
what does this migration files do? how is it used?

Reply:
`truffle migrate --network development` deploys all the satellites and controller, transfers ownership of the satellites to the controller and the unpauses the latter